### PR TITLE
[10.0] mig l10n_it_account_stamp

### PR DESCRIPTION
**Pull request by eLBati**
From https://github.com/OCA/l10n-italy/pull/706

### CODE REVIEW

File: l10n_it_account_stamp/models/invoice.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+    tax_stamp = fields.Boolean(
+        "Tax Stamp", readonly=True, states={'draft': [('readonly', False)]})
+
+    def is_tax_stamp_applicable(self):
+        stamp_product_id = self.env.user.with_context(
+            lang=self.partner_id.lang).company_id.tax_stamp_product_id
+        if not stamp_product_id:
+            raise exceptions.Warning(
```
Comment (SimoRubi) [inline l10n_it_account_stamp/models/invoice.py]:
```suggestion
            raise exceptions.UserError(
```

File: l10n_it_account_stamp/models/invoice.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        if self.is_tax_stamp_applicable():
+            self.tax_stamp = True
+        else:
+            self.tax_stamp = False
+
+    @api.multi
+    def add_tax_stamp_line(self):
+        for inv in self:
+            if not inv.tax_stamp:
+                raise exceptions.Warning(_("Tax stamp is not applicable"))
```
Comment (SimoRubi) [inline l10n_it_account_stamp/models/invoice.py]:
```suggestion
                raise exceptions.UserError(_("Tax stamp is not applicable"))
```

Comment (SimoRubi) [reply l10n_it_account_stamp/models/invoice.py]: Same for the other Warnings, should be substituted by `UserError`

File: l10n_it_account_stamp/models/invoice.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+                'account_id': stamp_account.id,
+                'price_unit': stamp_product_id.list_price,
+                'quantity': 1,
+                'uom_id': stamp_product_id.uom_id.id,
+                'invoice_line_tax_ids': [
+                    (6, 0, stamp_product_id.taxes_id.ids)],
+                'account_analytic_id': None,
+            })
+
+    def is_tax_stamp_line_present(self):
```
Comment (SimoRubi) [inline l10n_it_account_stamp/models/invoice.py]:
```suggestion
    @api.multi
    def is_tax_stamp_line_present(self):
        self.ensure_one()
```

File: l10n_it_account_stamp/models/invoice.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        for l in self.invoice_line_ids:
+            if l.product_id and l.product_id.is_stamp:
+                return True
+        return False
+
+    def _build_tax_stamp_lines(self, product):
```
Comment (SimoRubi) [inline l10n_it_account_stamp/models/invoice.py]:
```suggestion
    @api.multi
    def _build_tax_stamp_lines(self, product):
        self.ensure_one()
```

**Comment by SimoRubi** [comment]
@OCA/local-italy-maintainers could you please check? Thanks

File: l10n_it_fatturapa_out_stamp/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    "maintainers": ["eLBati"],
+    "license": "LGPL-3",
+    "application": False,
+    "installable": True,
+    "auto_install": True,
+    "depends": [
+        "l10n_it_fatturapa_out",
+        "l10n_it_account_stamp",
+    ],
+    "data": [
```
Comment (tafaRU) [inline l10n_it_fatturapa_out_stamp/__manifest__.py]: Please remove empty key

**Pull request merged**